spacetemplate/repository.go

```diff
@@ -0,0 +1,286 @@
+	for _, witHelper := range parsedSpaceTempl.WITs {
+		wit := witHelper.WorkItemType
+		var extendsTypePtr *uuid.UUID
+		if witHelper.ExtendsType != "" {
+			et, err := uuid.FromString(witHelper.ExtendsType)
+			if err != nil {
+				log.Info(ctx, map[string]interface{}{
+					"space_template_id": spaceTemplateID,
+					"wit_name":          wit.Name,
+					"space_id":          spaceID,
+					"err":               err,
+				}, "failed to parse work item type ID from string")
+				return errs.Wrapf(err, "failed to create work item type '%s' from space template '%s' in space '%s'", wit.Name, spaceTemplateID, spaceID)
+			}
+			extendsTypePtr = &et
+		}
+
+		witCreated, err := witRepo.Create(ctx, spaceID, &wit.ID, extendsTypePtr, wit.Name, wit.Description, wit.Icon, wit.Fields)
+		if err != nil {
+			log.Info(ctx, map[string]interface{}{
+				"space_template_id": spaceTemplateID,
+				"wit_name":          wit.Name,
+				"space_id":          spaceID,
+				"err":               err,
+			}, "failed to create work item type")
+			return errs.Wrapf(err, "failed to create work item type '%s' from space template '%s' in space '%s'", wit.Name, spaceTemplateID, spaceID)
+		}
+		if deleteArtifacts {
+			witToDelete := workitem.WorkItemType{ID: witCreated.ID}
+			defer r.db.Unscoped().Delete(witToDelete)
+		}
+	}
+
+	// Create WILTs
+	// Enfore all of them to be in the "user" link category
+	var linkCat *link.WorkItemLinkCategory
+	if len(parsedSpaceTempl.WILTs) > 0 {
+		linkCat, err = link.NewWorkItemLinkCategoryRepository(r.db).LoadCategoryFromDB(ctx, link.SystemWorkItemLinkCategoryUser)
+		if err != nil {
+			return errs.WithStack(err)
+		}
+	}
+	wiltRepo := link.NewWorkItemLinkTypeRepository(r.db)
+	for _, wiltHelper := range parsedSpaceTempl.WILTs {
+		if linkCat == nil {
+			return errs.Errorf("failed to dind work item link category: %s", link.SystemWorkItemLinkCategoryUser)
+		}
+		wilt := wiltHelper.WorkItemLinkType
+		wiltCreated, err := wiltRepo.Create(
+			ctx, wilt.Name,
+			wilt.Description,
+			wilt.SourceTypeID,
+			wilt.TargetTypeID,
+			wilt.ForwardName,
+			wilt.ReverseName,
+			wilt.Topology,
+			linkCat.ID,
+			spaceID)
+		if err != nil {
+			log.Info(ctx, map[string]interface{}{
+				"space_template_id": spaceTemplateID,
+				"wilt_name":         wilt.Name,
+				"space_id":          spaceID,
+				"err":               err,
+			}, "failed to create work item link type")
+			return errs.Wrapf(err, "failed to create work item link type '%s' from space template '%s' in space '%s'", wilt.Name, spaceTempl.ID, spaceID)
+		}
+		if deleteArtifacts {
+			wiltToDelete := link.WorkItemLinkType{ID: wiltCreated.ID}
+			defer r.db.Unscoped().Delete(wiltToDelete)
+		}
+	}
+
+	return nil
+}
+
+// createEmptySpaceTemplate can be used in tests in order to quickly create a
+// space template that doesn't do much except existing.
+func createEmptySpaceTemplate(ctx context.Context, repo Repository) (*SpaceTemplate, error) {
```

Comment:
Yes and no. It is used by the `spacetemplate.CreateEmptySpaceTemplate` function which is not only used by this package but also in the `controller` package. That's why I cannot put it in the tests.  `spacetemplate.CreateEmptySpaceTemplate` is also used in the `migration/migration.go` file.